packages/stream-api/src/streamApi.ts

```diff
@@ -77,7 +91,7 @@ export class StreamApi {
   ): Promise<vertexvis.protobuf.stream.IStreamResponse> {
     return new Promise(resolve => {
       const subscription = this.onResponse(response => {
-        if (response.frame != null) {
+        if (response.frame != null || response.hitItems) {
```

Comment:
This could be error prone, as you might have simultaneous requests for hit items. I think it'd be better to pass an optional request ID to `sendRequest` and switch the behavior if it's supplied to check that the response contains the request ID.

Reply:
@danschultz how does this look 6850723ebd175963696e2b934b4b214b696a5772?